### PR TITLE
{compiler,mpi,toolchain}[system/system] OpenMPI v4.1.1, nvompic v2021b, NVHPC v21.9

### PR DESCRIPTION
**Pull request by Micket**
(created using `eb --new-pr`)

This is just a reference to what it would look like if we try to make it similar to foss/2021a  w.r.t CUDA. 
But, we don't need to exactly follow

1. `NVHPC-21.7-CUDA-11.4.1.eb` +  `OpenMPI-4.1.1-NVHPC-21.7-CUDA-11.4.1.eb` + `nvompi-2021b-CUDA-11.4.1.eb` (like this PR) 
2. `NVHPC-21.7-CUDA-11.4.1.eb` +  `OpenMPI-4.1.1-NVHPC-21.7-CUDA-11.4.1.eb` + `nvompic-2021b.eb`
3. `NVHPC-21.7.eb` (still depending on CUDA-11.4.1.eb, but just skipping the suffix anyway) +  `OpenMPI-4.1.1-NVHPC-21.7.eb` + `nvompic-2021b.eb`

Having just made this PR, I actually think option 2 seems best.

I'm also not sure what the equivalent name for "foss/intel" would be here. `nvomkl` and `nvoolf`?

### CODE REVIEW

**Review by SebastianAchilles — CHANGES_REQUESTED** [review]
NVHPC 21.7 is not compatible to GCC 11. I suggest we wait for 21.9 which should be released soon.

**Comment by Micket** [comment]
preliminary change to 21.9, checksums to be updated of course. I also fixed a version mistake, and switched to option 2

File: easybuild/easyconfigs/n/NVHPC/NVHPC-21.9-CUDA-11.4.1.eb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+checksums = [
+    {
+        local_tarball_tmpl % 'aarch64':
+            '73eb3513845b59645f118b1e313472f54519dc252d5f5c32a05df2a2a8a19878',
+        local_tarball_tmpl % 'ppc64le':
+            '37ea23b5a9c696fb3fdb82855643afc4e02aea618102ec801206441f10fc9fba',
+        local_tarball_tmpl % 'x86_64':
+            '49d6e23492d131474698cf12971722d42e13a54a4eddec382e66e1053b4ac902',
+    }
+]
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/n/NVHPC/NVHPC-21.9-CUDA-11.4.1.eb:25]:
```suggestion
checksums = [
    {
        local_tarball_tmpl % 'aarch64':
            '52c2c66e30043add4afccedf0ba77daa0000bf42e0db844baa630bb635b91a7d',
        local_tarball_tmpl % 'ppc64le':
            'cff0b55fb782be1982bfeec1d9763b674ddbf84ff2c16b364495299266320289',
        local_tarball_tmpl % 'x86_64':
            '7de6a6880fd7e59afe0dee51f1fae4d3bff1ca0fb8ee234b24e1f2fdff23ffc9',
    }
]
```

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
rocky8-eb - Linux rocky linux 8.4, x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), Python 3.6.8
See https://gist.github.com/ee9c1a188d9c76a2d2f3b588d0215bcc for a full test report.

**Comment by SebastianAchilles** [comment]
> Test report by @SebastianAchilles
> **SUCCESS**
> Build succeeded for 1 out of 1 (1 easyconfigs in total)
> rocky8-eb - Linux rocky linux 8.4, x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), Python 3.6.8
> See https://gist.github.com/ee9c1a188d9c76a2d2f3b588d0215bcc for a full test report.

This is only `NVHPC-21.9-CUDA-11.4.1.eb`.

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 1 out of 3 (3 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/86dd710edad7b86fbcd4576185b854e6 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 1 out of 4 (4 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/87202dbbd4f67c033fdee4fbb06df831 for a full test report.

**Comment by Micket** [comment]
Hrm linking errors
```
ld: ./.libs/liblocal_ops_avx512.a(liblocal_ops_avx512_la-op_avx_functions.o):(.data+0x1250): multiple definition of `ompi_op_avx_3buff_functions_avx2'; ./.libs/liblocal_ops_avx2.a(liblocal_ops_avx2_la-op_avx_functions.o):(.data+0x1250): first defined here
ld: ./.libs/liblocal_ops_avx512.a(liblocal_ops_avx512_la-op_avx_functions.o):(.data+0x0): multiple definition of `ompi_op_avx_functions_avx2'; ./.libs/liblocal_ops_avx2.a(liblocal_ops_avx2_la-op_avx_functions.o):(.data+0x0): first defined here
```

**Review by jfgrimm — COMMENTED** [review]
As decided in issue #16330, we have deprecated the use of `True` to signify a system-toolchain dependency (#16384), in favour of the more intuitive `SYSTEM` template constant. Due to the change in the test suite, please run `eb --sync-pr-with-develop 14006` and update the PR to use `SYSTEM` instead.

**Pull request closed without merging**